arch/x86/compare258_avx512.c

```diff
@@ -0,0 +1,78 @@
+/* compare258_avx.c -- AVX2 version of compare258
```

Comment:
Should have proper filename and say AVX512. Your name should probably go in copyright.